app.py

```diff
@@ -0,0 +1,634 @@
+                                "padding": 10,
+                                "paddingLeft": 40,
+                                "paddingBottom": 60,
+                            },
+                        ),
+                    ],
+                    style={
+                        "outline": False,
+                        "className": "border-0 bg-transparent",
+                        "justify": "center",
+                    },
+                ),
+                dbc.Row(error_card),
+            ],
+            style={
+                "backgroundColor": "black",
+                "background-image": "url('assets/electric_squids.png')",
+                "background-size": "100%",
+                "paddingBottom": 50,
+            },
+            fluid=True,
+        ),
+    ]
+)
+
+
+@app.callback(
+    Output("input_employees", "value"),
+    Output("seed", "value"),
+    Output("checklist-input", "value"),
+    Output("min shifts", "value"),
+    Output("max shifts", "value"),
+    Output("shifts min", "value"),
+    Output("shifts max", "value"),
+    Output("cons shifts", "value"),
+    [Input("demo-dropdown", "value")],
+)
+def set_scenario(scenario_size):
+    if scenario_size == "Small":
+        return 12, 4, [2], 10, 20, 3, 6, 5
+    elif scenario_size == "Medium":
+        return 20, 4, [2], 8, 16, 6, 10, 5
+    elif scenario_size == "Large":
+        return 40, 4, [2], 4, 16, 6, 10, 5
+    else:
+        return 12, None, [2], 10, 20, 3, 6, 5
+
+
+@app.callback(
+    Output("initial-sched", "children"),
+    [Input("input_employees", "value"), Input("seed", "value")],
+)
+def disp_initial_sched(*vals):
+    if vals:
+        num_employees = vals[0] - 1
+        rand_seed = vals[1]
+    else:
+        num_employees = 9  # one less to account for trainee
+
+    now = datetime.now()
+    month = now.month
+    year = now.year
+    num_days = calendar.monthrange(year, month)[1]
+
+    shifts = [str(i + 1) for i in range(num_days)]
+    df = utils.build_random_sched(num_employees, shifts, rand_seed)
+
+    df.replace(False, " ", inplace=True)
+    df.replace(True, "X", inplace=True)
+
+    return utils.display_availability(df, month, year)
+
+
+# Don't allow max shifts to be smaller than min shifts
+@app.callback(Output("max shifts", "min"), Input("min shifts", "value"))
+def set_min_shifts(min_s):
+    return min_s
+
+
+# Don't allow min shifts to be bigger than max shifts
+@app.callback(Output("shifts min", "max"), Input("shifts max", "value"))
+def set_max_shifts(max_s):
+    return max_s
+
+
+# Don't allow max employees to be smaller than min employees
+@app.callback(Output("shifts max", "min"), Input("shifts min", "value"))
+def set_shifts_min(min_e):
+    return min_e
+
+
+# Don't allow min employees to be bigger than max employees
+@app.callback(Output("min shifts", "max"), Input("max shifts", "value"))
+def set_shifts_max(max_e):
+    return max_e
+
+
+@app.callback(
+    Output("built-sched", "children"),
```

Comment:
Currently, changing inputs such as that the Employee Availability schedule changes does not blank out an existing Employee Schedule. So one gets a mismatch:

 
![image](https://github.com/dwave-examples/employee-scheduling/assets/34041130/b652e8cc-2d3c-4af4-bf63-6cdb0fc4e0c8)
![image](https://github.com/dwave-examples/employee-scheduling/assets/34041130/f8bb6c3e-6f82-45bb-b526-79c0784a7357)

Be nice to blank out the existing result when the user changes the inputs.